README.md

```diff
@@ -24,7 +24,125 @@ $ helm install nfs-subdir-external-provisioner nfs-subdir-external-provisioner/n
     --set nfs.path=/exported/path
 ```
 
-### Without Helm
+### With Kustomize
+
+**Step 1: Get connection information for your NFS server**
+
+Make sure your NFS server is accessible from your Kubernetes cluster and get the information you need to connect to it. At a minimum you will need its hostname and exported share path.
+
+**Step 2: Add the base resource**
+
+Create a `kustomization.yaml` file in a directory of your choice, and add the [deploy](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/tree/master/deploy) directory as a resource. This will use the kustomization file within that directory as our base.
+
+```yaml
+namespace: nfs-provisioner
+resources:
+  - github.com/kubernetes-sigs/nfs-subdir-external-provisioner//deploy
+```
+
+**Step 3: Create namespace resource**
+
+Create a file with your namespace resource. The name can be anything as it will get overwritten by the namespace in your kustomization file.
+
+```yaml
+# namespace.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: nfs-provisioner
+```
+
+**Step 4: Configure deployment**
+
+To configure the deployment, you will need to patch it's container variables with the connection information for your NFS Server.
+
+```yaml
+# patch_nfs_details.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nfs-client-provisioner
+  name: nfs-client-provisioner
+spec:
+  template:
+    spec:
+      containers:
+        - name: nfs-client-provisioner
+          env:
+            - name: NFS_SERVER
+              value: <YOUR_NFS_SERVER_IP>
+            - name: NFS_PATH
+              value: <YOUR_NFS_SERVER_SHARE>
+      volumes:
+        - name: nfs-client-root
+          nfs:
+            server: <YOUR_NFS_SERVER_IP>
+            path: <YOUR_NFS_SERVER_SHARE>
+```
+
+Replace occurrences of `<YOUR_NFS_SERVER_IP>` and `<YOUR_NFS_SERVER_SHARE>` with your connection information.
+
+**Step 5: Add resources and deploy**
+
+Add the namespace resource and patch you created in earlier steps.
+
+```yaml
+namespace: nfs-provisioner
+resources:
+  - github.com/kubernetes-sigs/nfs-subdir-external-provisioner//deploy
+  - namespace.yaml
+patchesStrategicMerge:
+  - patch_nfs_details.yaml
+```
+
+Deploy (run inside directory with your kustomization file):
+
+```sh
+kubectl apply -k .
```

Comment:
it's still not working with kubectl v1.18.8:

```console
$ cat kustomization.yaml 
namespace: nfs-provisioner
resources:
  - github.com/Weilbyte/nfs-subdir-external-provisioner//deploy?ref=Weilbyte/kustomize
  - namespace.yaml
patchesStrategicMerge:
  - patch_nfs_details.yaml
```

```console
$ kubectl apply -k .
error: rawResources failed to read Resources: Load from path github.com/Weilbyte/nfs-subdir-external-provisioner//deploy?ref=Weilbyte/kustomize failed: evalsymlink failure on '/home/yonatan/pr90/github.com/Weilbyte/nfs-subdir-external-provisioner/deploy?ref=Weilbyte/kustomize' : lstat /home/yonatan/pr90/github.com: no such file or directory
```

i don't know on which version the remote git url feature introduced, but could be that it is available only in kubectl>=v1.21 (see ref1 below) and when i tried with kubectl v1.21.2 it worked fine. 
i think we need to mention this requirement in the beginning. 
OR, and also to support wider range of versions and potential users, we can tell them to clone the git repo and just use the resource locally.
wdyt?


ref1:
| Kubectl version | Kustomize version |
| --- | --- |
| < v1.14 | n/a |
| v1.14-v1.20 | v2.0.3 |
| v1.21 | v4.0.5 |

Reply:
Both of the approaches can work but I personally think it would be simpler to use the git url. But kubectl >=v1.21 does appear to be required for that.

I am not able to test right now but maybe `git@github.com:Weilbyte/nfs-subdir-external-provisioner//deploy?ref=Weilbyte/kustomize` might work for under v1.21?

Reply:
unfortunately none of these options worked in kubectl v1.18.8

Reply:
I am sorry for taking long to reply, I have not received a notification. An alternative might be just instructing the user to download Kustomize standalone since the versions packaged in the most commonly used kube versions appear to be ancient. Then just feed the output of kustomize to kubectl apply.